src/opm/output/eclipse/FileService/EclOutput.cpp

```diff
@@ -217,6 +231,41 @@ void EclOutput::writeBinaryCharArray(const std::vector<std::string>& data)
     }
 }
 
+void EclOutput::writeBinaryCharArray(const std::vector<Opm::RestartIO::Helpers::CharArrayNullTerm<8>>& data)
+{
+    const auto size = data.size();
+
+    const auto sizeData = EIOD::block_size_data_binary(EIOD::CHAR);
+
+    const int sizeOfElement       = std::get<0>(sizeData);
+    const int maxBlockSize        = std::get<1>(sizeData);
+    const int maxNumberOfElements = maxBlockSize / sizeOfElement;
+
+    int rest = size * sizeOfElement;
+
+    if (!ofileH.is_open()) {
```

Comment:
The file handle is checked on construction - that should be enough!

Reply:
> The file handle is checked on construction - that should be enough!

Technically introduced in #699.

Reply:
I know; but know it is merged - and it is time to improve on things.